packages/tools/viewer/src/viewer.ts

```diff
@@ -1466,6 +1541,330 @@ export class Viewer implements IDisposable {
+            }
+
+            const shadowMaterial = new ShadowOnlyMaterial("mat", this._scene);
+            const ground = this._createClassicShadowGround();
+            ground.material = shadowMaterial;
+            this._shadowLight.includedOnlyMeshes = [ground];
+        }
+
+        // manually set the extends to take into account animated meshes
+        if (this._shadowLight) {
+            this._shadowLight.position.set(x * radius, radius, z * radius);
+            this._shadowLight.autoUpdateExtends = false;
+            this._shadowLight.shadowMinZ = 0.000000007;
+            this._shadowLight.shadowMaxZ = radius * 4;
+            this._shadowLight.orthoLeft = -radius * 2;
+            this._shadowLight.orthoRight = radius * 2;
+            this._shadowLight.orthoTop = radius * 2;
+            this._shadowLight.orthoBottom = -radius * 2;
+        }
+
+        const groundSize = this._shadowGroundScalingFactor * radius;
+        this._classicShadowGround?.scaling.set(groundSize, groundSize, groundSize);
+
+        this._loadedModelsBacking.forEach((model) => {
+            const mesh = model.assetContainer.meshes[0];
+            this._shadowGenerator?.addShadowCaster(mesh, true);
+            model.assetContainer.meshes.forEach((mesh) => {
+                mesh.receiveShadows = true;
+            });
+        });
+
+        if (this._classicShadowGround) {
+            this._classicShadowGround.position.y = worldBounds.extents.min[1];
+        }
+
+        this._envShadowGround?.setEnabled(false);
+        this._iblShadowsRenderPipeline?.toggleShadow(false);
+        this._classicShadowGround?.setEnabled(true);
+
+        this._snapshotHelper.enableSnapshotRendering();
+        this._markSceneMutated();
+    }
+
+    private _disposeShadows() {
+        this._snapshotHelper.disableSnapshotRendering();
+
+        this._loadedModelsBacking.forEach((model) => {
+            const meshes = model.assetContainer.meshes as Mesh[];
+
+            const mesh = model.assetContainer.meshes[0];
+            this._shadowGenerator?.removeShadowCaster(mesh, true);
+            mesh.receiveShadows = false;
+
+            meshes.forEach((mesh) => {
+                this._iblShadowsRenderPipeline?.removeShadowCastingMesh(mesh);
+                if (mesh.material) {
+                    this._iblShadowsRenderPipeline?.removeShadowReceivingMaterial(mesh.material);
+                }
+            });
+        });
+
+        this._shadowGenerator?.dispose();
+        this._shadowLight?.dispose();
+        this._shadowGenerator = null;
+        this._shadowLight = null;
+
+        this._resizeShadowObserver?.remove();
+        this._groundShadowMaterial?.dispose();
+        this._iblShadowsRenderPipeline?.dispose();
+        this._iblShadowsRenderPipeline = null;
+
+        this._classicShadowGround?.dispose();
+        this._classicShadowGround = null;
+        this._envShadowGround?.dispose();

```

Comment:
Same thing, if we pass the flag to dispose materials, we don't need a separate call to explicitly dispose the associated material, right?